### PR TITLE
Added CPU limits for deployments

### PR DESCRIPTION
**Pull request by nicholasjackson**
Added CPU limits for deployments
Added capability to pass CPU based constraint with a stackfile

[https://github.com/openfaas/faas-cli/issues/250](https://github.com/openfaas/faas-cli/issues/250)

## Description
The stack schema has been changed adding yaml struct tags and also the deployment proxy has been modified to pass the CPU based constraint.

## Motivation and Context
CPU and memory based constraints should be possible
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Tested with local build and deployment to k8s

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
@nicholasjackson can you edit the title please?

**Comment by nicholasjackson** [comment]
@alexellis done, I messed my commit message, have changed

File: proxy/deploy.go [hunk 1]
```diff
@@ -49,16 +50,23 @@ func DeployFunction(fprocess string, gateway string, functionName string, image
 		Labels:      &labels,
 	}
 
+	req.Limits = &requests.FunctionResources{}
```
Comment (alexellis) [inline proxy/deploy.go]: We used to pass `nil`, but now always pass an initialized object?

File: stack/schema.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ type Function struct {
 	Environment map[string]string `yaml:"environment"`
 
 	// Secrets list of secrets to be made available to function
-	Secrets []string `json:"secrets"`
+	Secrets []string `yaml:"secrets"`
```
Comment (alexellis) [inline stack/schema.go]: Good change 👍

File: proxy/deploy.go (hunk 1 above)
Comment (alexellis) [reply proxy/deploy.go]: Probably best to only initialise payload if we think we have a value.

**Comment by alexellis** [comment]
Just checked FaaS repo - this doesn't apply to Docker Swarm then..?

**Pull request merged**